persistence/src/main/scala/com/mesosphere/usi/repository/RecordRepository.scala

```diff
@@ -14,33 +14,23 @@ trait RecordRepository {
     * @param record
     * @return id of the provided record
     */
-  def create(record: Record): Future[RecordId]
+  def store(record: Record): Future[RecordId]
 
   /**
-    * Retrieves the record if it exists.
-    * @param recordId
-    * @return Option(record) if it exists, None otherwise
-    */
-  def read(recordId: RecordId): Future[Option[Record]]
-
-  /**
-    * Updates the record if it exists. If the record is missing,
-    * the future will be failed with an [[RecordNotFoundException]].
-    *
-    * @param record
-    * @return id of the updated record
+    * Retrieves all the existing records (if any).
+    * @return Map containing all the current pod records. Can be empty if there are no pod records.
     */
-  def update(record: Record): Future[RecordId]
+  def readAll(): Future[Map[RecordId, Record]]
```

Comment:
Should we provide a stream instead to leverage backpressure?

Reply:
Current structure assumes we are intentionally blocking inlets until the state can be loaded. Based on the outcome of that structure, we can chose to change this to be a stream instead.

Reply:
Where are we blocking? I must be missing something?